### ISSUE TITLE
`authorize()` should check if cached credentials already exist

### ISSUE THREAD

**Issue by howardbaek** [fhdsl/metricminer#48]
Currently, `authorize()` doesn't check if the user already has cached tokens (sometime in the past) and if they do, then don't show these two questions:

<img width="745" alt="Screenshot 2023-12-06 at 10 56 00 AM" src="https://github.com/fhdsl/metricminer/assets/50791792/36fe017e-52a9-4647-9c1c-a1873be91744">

I think we should check for tokens in the cache folder and if it doesn't exist, then display the above questions, but if it does exist, then grab those cached tokens.

**Comment by cansavvy** [comment]
Yes. This has been a thing I keep noticing but have yet to do anything about it. Thanks for tracking it @howardbaek
